From 417fe1f3032416dc3ff62aa568d5e5d1a1881d5c Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:17 -0400
Subject: [ethernet] i40e: set lan_veb index

Message-id: <1409658042-22689-137-git-send-email-sassmann@redhat.com>
Patchwork-id: 90331
O-Subject: [RHEL7.1 PATCH 136/281] i40e: set lan_veb index
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

When the VEB is created for the basic LAN device and its VSIs, we need to
set the tracking lan_veb index for later use.

Change-ID: I66bb74993bbda3621ca557437cb4b3517f9b315b
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 1bb8b9358ee0f6b297c4f33ca0a3a45338b5205c)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index cf14f66..6cf25a6 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -7735,6 +7735,8 @@ struct i40e_veb *i40e_veb_setup(struct i40e_pf *pf, u16 flags,
  ret = i40e_add_veb(veb, pf->vsi[vsi_idx]);
  if (ret)
   goto err_veb;
+ if (vsi_idx == pf->lan_vsi)
+  pf->lan_veb = veb->idx;
 
  return veb;
 
-- 
1.7.1